pkg/controller/context.go

```diff
@@ -322,21 +323,28 @@ func NewContextFactory(ctx context.Context, opts ContextOptions) (*ContextFactor
 func (c *ContextFactory) Build(component ...string) (*Context, error) {
 	restConfig := util.RestConfigWithUserAgent(c.baseRestConfig, component...)
 
+	scheme := runtime.NewScheme()
+	for _, err := range []error{
+		kscheme.AddToScheme(scheme),
+		cmscheme.AddToScheme(scheme),
+		gwscheme.AddToScheme(scheme),
+	} {
+		if err != nil {
+			return nil, fmt.Errorf("error adding scheme to runtime: %w", err)
+		}
+	}
```

Comment:
This will only show the first error. 
Use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors#NewAggregate ,
or explain why you think a for loop is a better solution here.